packages/neon/neon/lib/src/widgets/error.dart

```diff
@@ -118,21 +118,10 @@ class NeonError extends StatelessWidget {
       );
     }
 
-    if (error is MissingPermissionException) {
+    if (error is NeonException) {
```

Comment:
Why don't let the neon exception provide the `ExceptionDetails` directly?

Reply:
Right, but we still need to pass in a context in order to have the localizations since exceptions can be thrown from blocs too where we don't have a context available.

Reply:
You could make `ExceptionDetails.text` of type `LabelBuilder` (and probably rename the field) so the context would only need to be given at display time.